### PR TITLE
Fix #3766 - add other matching variants for SVs

### LINKED ISSUE #3766: Matching causatives on cancer SV variant page
Hej, for Cancer Scout, if a SV is found as matching causative (on start page), we would like to see the case ID of that other case/other cases (in the variant page). For SNVs we have a list with cases in which the variant has been marked as causative.

_Originally posted by @christacosta in https://github.com/Clinical-Genomics/scout/issues/3719#issuecomment-1385438750_

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality or fixes a bug.

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. find an SV with a matching causative (or create one by marking an SV shared between two cases causative), e.g. https://scout-stage.scilifelab.se/cust002/F0013364/sv/variants/7df95a2accbbba110c75062a7dc1d8b5
2. see that the other variant is not displayed on main
3. apply patch and see the matching causative displayed on the variant page.
4. Ideally rinse and repeat for somatic

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by CR
- [x] tests executed by DN

### CODE REVIEW

**Comment by dnil** [comment]
☑️ Looks right on stage now:
Screenshot 2023-01-17 at 15 34 57

**Comment by dnil** [comment]
☑️ Looks right on stage now:
Screenshot 2023-01-17 at 15 34 57

File: scout/server/blueprints/variant/templates/variant/sv-variant.html [hunk 1]
```diff
@@ -67,8 +67,18 @@
...
     </div>
-
   {% endif %}
+
+   <div class="row">
+     <div class="col-12">
+       {% if variant.category == "cancer_sv" %}
+        {{matching_variants(managed_variant, causatives, variant.matching_tiered) }}
+       {% else %}
+        {{matching_variants(managed_variant, causatives) }}
```
Comment (northwestwitch) [inline scout/server/blueprints/variant/templates/variant/sv-variant.html:77]: Why invoking twice this macro?

Comment (dnil) [reply scout/server/blueprints/variant/templates/variant/sv-variant.html:77]: Where is the other invokation? Or you mean like why different for cancer and RD?
If the latter, I'm not 100% how unpopulated the matching_tired would be for an RD case. Do you know?

Comment (northwestwitch) [reply scout/server/blueprints/variant/templates/variant/sv-variant.html:77]: matching_variants is a macro from components.html. You call it twice (line 75 and 77)

Comment (northwestwitch) [reply scout/server/blueprints/variant/templates/variant/sv-variant.html:77]: > I'm not 100% how unpopulated the matching_tired would be for an RD case. Do you know?

Nope!

Comment (northwestwitch) [reply scout/server/blueprints/variant/templates/variant/sv-variant.html:77]: Ah no, sorry, I misread! Still half asleep! 😆

**Review by northwestwitch — APPROVED** [review]
Looks good, if you pay attention to the IF ELSE 🤣 

💯

File: scout/server/blueprints/variant/templates/variant/sv-variant.html (hunk 1 above)
Comment (dnil) [reply scout/server/blueprints/variant/templates/variant/sv-variant.html:77]: Hihi, mornings! Best time for some coffee!

**Pull request merged**